_posts/2025-04-23-secure-mcp-sse-server.adoc

```diff
@@ -0,0 +1,202 @@
+---
+layout: post
+title: 'Getting ready for secure MCP with Quarkus MCP Server'
+date: 2025-04-23
+tags: ai mcp security
+synopsis: 'Demonstrate secure Quarkus MCP SSE server access in the dev mode'
+author: sberyozkin
+---
+:imagesdir: /assets/images/posts/secure_mcp_sse_server
+
+== Introduction
+
+https://modelcontextprotocol.io/specification/2025-03-26[The latest version of the Model Context Protocol (MCP) specification] introduces an https://modelcontextprotocol.io/specification/2025-03-26/basic/authorization[authorization] flow for HTTP based MCP transports.
+
+MCP authorization will enable MCP SSE and Streamable HTTP clients to access MCP servers securely and is being very actively discussed and evaluated right now. This https://github.com/modelcontextprotocol/modelcontextprotocol/pull/284[MCP authorization PR] is one of the focal points for the MCP authorization discussions.
+
+While it will take a bit of time for the MCP authorization specificaion to finalize and be widely supported, one thing you can be certain about is that secure MCP servers that wish to interoperate with MCP authorization aware MCP clients will be required to accept and verify bearer access tokens sent by these clients on behalf of their users.
```

Comment:
Here, there is an issue - so far (except if I missed it), this is not part of the capabilities of the server. Thus, the client must now that the server requires authentication. Also, I'm not sure about multi-users, in the sense that the MCP client may have a "connection" (which is actually a client in the spec) with a server. It's unclear if these can be used by multiple users, or if you need to instantiate one "client" per user.